test/e2e/antreapolicy_test.go

```diff
@@ -680,6 +681,59 @@ func testACNPDropIngressInSelectedNamespace(t *testing.T) {
 	executeTests(t, testCase)
 }
 
+func testACNPDropIPBlockWithExcept(t *testing.T) {
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	builder = builder.SetName("acnp-drop-all-egress-from-ya-except-xa-xb-ip").
+		SetPriority(1.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{PodSelector: map[string]string{"pod": "a"}, NSSelector: map[string]string{"ns": getNS("y")}}})
+	podXAIP, _ := podIPs[getPodName("x", "a")]
+	podXBIP, _ := podIPs[getPodName("x", "b")]
+	ipBlocks := genIPBlockWithExceptIPs(append(podXAIP, podXBIP...))
+	for i := range ipBlocks {
+		builder.AddEgress(ProtocolTCP, &p80, nil, nil, nil, nil, nil, nil, ipBlocks[i], nil, nil, nil, nil, nil, nil, nil, nil,
+			crdv1beta1.RuleActionDrop, "", "egress-drop-"+strconv.Itoa(i), nil)
+	}
+	reachability := NewReachability(allPods, Connected)
+	reachability.ExpectAllEgress(getPod("y", "a"), Dropped)
+	reachability.Expect(getPod("y", "a"), getPod("x", "a"), Connected)
+	reachability.Expect(getPod("y", "a"), getPod("x", "b"), Connected)
+	reachability.Expect(getPod("y", "a"), getPod("y", "a"), Connected)
+	testStep := []*TestStep{
+		{
+			Name:          "Port 80",
+			Reachability:  reachability,
+			TestResources: []metav1.Object{builder.Get()},
+			Ports:         []int32{80},
+			Protocol:      ProtocolTCP,
+		},
+	}
+	builder2 := &ClusterNetworkPolicySpecBuilder{}
+	builder2 = builder2.SetName("acnp-drop-egress-from-ya-to-xa").
+		SetPriority(2.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{PodSelector: map[string]string{"pod": "a"}, NSSelector: map[string]string{"ns": getNS("y")}}}).
+		AddEgress(ProtocolTCP, &p80, nil, nil, nil, nil, nil, nil, nil, map[string]string{"pod": "a"}, nil, map[string]string{"ns": getNS("x")}, nil, nil, nil, nil, nil,
+			crdv1beta1.RuleActionDrop, "", "egress-drop-xa", nil)
+	// Make sure that the except IPs in the previous policy excluded from the drop rule but not explicitly allowed
```

Comment:
I don't think I understand this comment
as far as I can tell, there is nothing "explicitly allowed", and we are just adding an extra drop rule to supplement the previous one and drop one extra connection (y/a -> x/a)

Reply:
Rephrased